components/identities/actions/addEvmAddrModal.tsx

```diff
@@ -0,0 +1,234 @@
+
+  const { writeAsync: set_user_data, data: userData } = useContractWrite({
+    calls:
+      identity && isValid
+        ? [
+            identityChangeCalls.setUserData(
+              identity.id,
+              shortString.encodeShortString("evm-address"),
+              hexToDecimal(evmAddress)
+            ),
+          ]
+        : [],
+  });
+
+  useEffect(() => {
+    // Reset the input field when the modal is closed
+    setEvmAddress(identity?.evmAddress);
+    setFieldInput(identity?.evmAddress ?? "");
+    setIsValid(true);
+    setMessage(undefined);
+  }, [isModalOpen]);
+
+  useEffect(() => {
+    if (!userData?.transaction_hash) return;
+    addTransaction({
+      timestamp: Date.now(),
+      subtext: `EVM addressed updated for ${identity?.domain}`,
+      type: NotificationType.TRANSACTION,
+      data: {
+        type: TransactionType.SET_USER_DATA,
+        hash: userData.transaction_hash,
+        status: "pending",
+      },
+    });
+    setIsTxSent(true);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [userData]); // We want to execute this only once when the tx is sent
+
+  function setUserData(): void {
+    set_user_data();
+  }
+
+  function changeAddress(value: string): void {
+    setFieldInput(value);
+    isValidAddrOrName(value);
+  }
+
+  function isValidAddrOrName(addressOrName: string): void {
+    setLoading(true);
+
+    // Cancel the previous request if it's still pending
+    if (abortControllerRef.current) {
+      abortControllerRef.current.abort();
+    }
+
+    // Create a new AbortController for the new request
+    const abortController = new AbortController();
+    abortControllerRef.current = abortController;
+
+    if (isValidEns(addressOrName)) {
+      fetch(`https://enstate.rs/n/${addressOrName}`, {
+        signal: abortController.signal,
+      })
+        .then((res) => res.json())
+        .then((data) => {
+          if (data.address) {
+            setEvmAddress(data.address);
+            setMessage(`Resolving to ${data.address}`);
+            setIsValid(true);
+          } else {
+            setEvmAddress(undefined);
+            setMessage("Invalid ENS name");
+            setIsValid(false);
+          }
+          setLoading(false);
+        })
+        .catch((error) => {
+          // Request was aborted
+          if (error.name === "AbortError") return;
+          setEvmAddress(undefined);
+          setIsValid(false);
+          setMessage("Invalid address or ENS");
+          setLoading(false);
+        });
+    } else if (ethers.isAddress(addressOrName)) {
+      setEvmAddress(addressOrName);
+      setIsValid(true);
+      setMessage(undefined);
+      setLoading(false);
+    } else {
+      setEvmAddress(undefined);
+      setIsValid(false);
+      setMessage("Invalid address or ENS");
+      setLoading(false);
+    }
+  }
+
+  function closeModal(showNotif: boolean): void {
+    setIsTxSent(false);
+    handleClose(showNotif);
```

Comment:
That's what I was doing in the useEffect : 

```
  useEffect(() => {
    // Reset the input field when the modal is closed
    setEvmAddress(identity?.evmAddress);
    setFieldInput(identity?.evmAddress ?? "");
    setIsValid(true);
    setMessage(undefined);
  }, [isModalOpen]);
```

It might make more sens to remove the `useEffect`, and reset everything when `closeModal` is called.

Reply:
Feedback integrated